### PR TITLE
fix(parental-leave): minDate when picking new period

### PR DESCRIPTION
**Pull request by jeremybarbet**
## What

Min date when picking new periods dates

https://user-images.githubusercontent.com/937328/119796146-a4a5d480-bec8-11eb-8ed7-8024d0602f01.mov

### CODE REVIEW

**Review by baering — COMMENTED** [review]
Nice, thanks for tackling this 👍  I left some points to discuss.

File: libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts [hunk 1]
```diff
@@ -777,13 +777,15 @@ export const ParentalLeaveForm: Form = buildForm({
...
-                  },
+                  minDate: (application) =>
+                    addDays(
+                      getEndPeriodsDate(
+                        application.answers.periods as Period[],
+                      ),
+                      1,
+                    ),
+                  excludeDates: (application) =>
+                    getAllPeriodDates(application.answers.periods as Period[]),
```
[the remark this answers is not in the document]
Comment (jeremybarbet) [reply libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts:788]: Good idea, a nice improvement for later on

File: libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import addDays from 'date-fns/addDays'
```
[the remark this answers is not in the document]
Comment (jeremybarbet) [reply libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts:1]: We will have the VMST API for it soon (likely this week) so we can replace all the date-fns with the new utility :D

File: libs/application/templates/parental-leave/src/parentalLeaveUtils.ts [hunk 3]
```diff
@@ -193,6 +194,26 @@ export const getAllPeriodDates = (periods: Period[]) => {
...
+  return filledPeriods.reduce((acc, cur) => {
+    const date = new Date(cur.endDate)
+
+    if (date > acc) {
+      acc = date
+    }
+
+    return acc
+  }, new Date(0, 0, 0))
+}
```
[the remark this answers is not in the document]
Comment (jeremybarbet) [reply libs/application/templates/parental-leave/src/parentalLeaveUtils.ts:209]: Good point, I didn't think of that, I will remove the minDate for the startDate then 👍

**Review by madebynoam — APPROVED** [review]
Lgtm!

**Pull request merged**